test/local/mysql_tests.js

```diff
@@ -1,388 +1,369 @@
-              t.equal(Object.keys(results[1][0]).length, 1, 'second result item had one property')
-              t.ok(results[1][0].count >= 0, 'count property was non-negative number')
+            function(result) {
+              assert(false, 'this query should never succeed!')
+            },
+            function(err) {
+              assert(true, 'we got an error')
+              assert.equal(err.code, 500)
+              assert.equal(err.errno, 1146)
+              assert.equal(err.error, 'Internal Server Error')
+              assert.equal(err.message, 'ER_NO_SUCH_TABLE')
+              throw err
             }
           )
-        }
-      )
+      }
 
-      test(
-        'readMultiple with final query',
-        function (t) {
-          t.plan(1)
-          return db.readMultiple([
-            { sql: 'SELECT * FROM accounts LIMIT 1' },
-            { sql: 'SELECT * FROM accounts LIMIT 1' }
-          ], { sql: 'SELECT * FROM accounts LIMIT 1' })
-          .then(
-            function(results) {
-              t.equal(results.length, 2, 'results array contained two items')
-            }
-          )
-        }
-      )
+      return db.retryable_(writer, [ 1146 ])
+        .then(
+          function(result) {
+            assert(false, 'This should never happen, even with a retry ' + callCount)
+          },
+          function(err) {
+            assert.equal(callCount, 2, 'the function was retried')
+          }
+        )
+    }
+  )
 
-      test(
-        'readMultiple with error in query',
-        function (t) {
-          t.plan(1)
-          return db.readMultiple([
-            { sql: 'SELECT nonsense FROM gibberish' },
-            { sql: 'SELECT * FROM accounts LIMIT 1' }
-          ], { sql: 'SELECT * FROM accounts LIMIT 1' })
-          .then(
-            function(results) {
-              t.fail('should have failed')
-            },
-            function(err) {
-              t.pass('failed correctly')
+  it(
+    'check that an error in a stored procedure (with transaction) is propagated back',
+    () => {
+      // let's add a stored procedure which will cause an error
+      var dropProcedure = 'DROP PROCEDURE IF EXISTS `testStoredProcedure`;'
+      var ensureProcedure = [
+        'CREATE PROCEDURE `testStoredProcedure` ()',
+        'BEGIN',
+        '    DECLARE EXIT HANDLER FOR SQLEXCEPTION',
+        '    BEGIN',
+        '        -- ERROR',
+        '        ROLLBACK;',
+        '        RESIGNAL;',
+        '    END;',
+        '    START TRANSACTION;',
+        '    INSERT INTO accounts(uid) VALUES(null);',
+        '    COMMIT;',
+        'END;',
+      ].join('\n')
+
+      return db.write(dropProcedure, [])
+        .then(function() {
+          assert(true, 'Drop procedure was successful')
+          return db.write(ensureProcedure, [])
+        })
+        .then(
+          function(result) {
+            assert(true, 'The stored procedure creation was successful')
```

Comment:
I think I prefer it when you just assert the string message instead of `true`, like you do in `test/local/prune_tokens.js`.